src/platform/Ameba/ConfigurationManagerImpl.cpp

```diff
@@ -88,12 +88,29 @@ CHIP_ERROR ConfigurationManagerImpl::Init()
     err = Internal::GenericConfigurationManagerImpl<AmebaConfig>::Init();
     SuccessOrExit(err);
 
-    // If the fail-safe was armed when the device last shutdown, initiate a factory reset.
+    // If the fail-safe was armed when the device last shutdown, initiate cleanup to the pending Fail Safe Context with
+    // which the fail-safe timer has been armed.
     if (GetFailSafeArmed(failSafeArmed) == CHIP_NO_ERROR && failSafeArmed)
     {
-        ChipLogProgress(DeviceLayer, "Detected fail-safe armed on reboot; initiating factory reset");
-        InitiateFactoryReset();
+        FabricIndex fabricIndex;
+        bool addNocCommandInvoked;
+        bool updateNocCommandInvoked;
+
+        ChipLogProgress(DeviceLayer, "Detected fail-safe armed on reboot");
+
+        err = FailSafeContext::LoadFromStorage(fabricIndex, addNocCommandInvoked, updateNocCommandInvoked);
+        SuccessOrExit(err);
+
+        ChipDeviceEvent event;
+        event.Type                                                = DeviceEventType::kFailSafeTimerExpired;
+        event.FailSafeTimerExpired.PeerFabricIndex                = fabricIndex;
+        event.FailSafeTimerExpired.AddNocCommandHasBeenInvoked    = addNocCommandInvoked;
+        event.FailSafeTimerExpired.UpdateNocCommandHasBeenInvoked = updateNocCommandInvoked;
+
+        err = PlatformMgr().PostEvent(&event);
+        SuccessOrExit(err);
```

Comment:
But ....

1. Should we not clear the `failSafeArmed` boolean after cleanup is complete?
2. Should we not clear the serialized failsafe state from storage after the cleanup is complete?

Otherwise on our next boot we will try to do this cleanup again....